### PR TITLE
Feat/quiz routing

### PR DESCRIPTION
**Pull request by Zenteh-NTU**
- Added some styling for responsive UI. 
- Moved Question Banks as Quiz. 
- Fixed Radio Group bug on creating AI generated question bank.

### CODE REVIEW

File: apps/web/src/app/dashboard/teacher/quiz/[slug]/_components/NewQuestionButton.tsx [hunk 1]
```diff
@@ -92,7 +92,18 @@ export const NewQuestionButton = ({
...
+      }
+      if (generated.answers[1].isCorrect) {
+        form.setValue("correct", "2");
+      }
+      if (generated.answers[2].isCorrect) {
+        form.setValue("correct", "3");
+      }
+      if (generated.answers[3].isCorrect) {
+        form.setValue("correct", "4");
+      }
```
Comment (RealDyllon) [inline apps/web/src/app/dashboard/teacher/quiz/[slug]/_components/NewQuestionButton.tsx]: this seems a bit verbose. maybe we could do smth like

```typescript
// set correct radio button
form.setValue("correct", (form.answers.findIndex(answer => answer.isCorrect) + 1).toString());

```

File: apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+export const QuestionBankCards = ({ initialData }: Props) => {
+  const { data: questionsBanks, isLoading } = trpc.questionBank.list.useQuery(
+    undefined,
+    {
+      initialData
+    }
+  );
+  console.log(questionsBanks);
```
Comment (RealDyllon) [inline apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx]: can remove

File: apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            {/*  viewBox="0 0 24 24"*/}
+            {/*  fill="none"*/}
+            {/*  stroke="currentColor"*/}
+            {/*  strokeLinecap="round"*/}
+            {/*  strokeLinejoin="round"*/}
+            {/*  strokeWidth="2"*/}
+            {/*  className="h-4 w-4 text-muted-foreground"*/}
+            {/*>*/}
+            {/*  <path d="M12 2v20M17 5H9.5a3.5 3.5 0 0 0 0 7h5a3.5 3.5 0 0 1 0 7H6" />*/}
+            {/*</svg>*/}
```
Comment (RealDyllon) [inline apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx]: can remove commented out code

File: apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                  View
+                </Button>
+              </Link>
+              <NewQuizButton initialData={questionBank}></NewQuizButton>
+            </div>
+          </CardFooter>
+          {/*<CardFooter className="flex flex-row justify-between">*/}
+          {/*  <EditQuestionBankButton initialData={questionBank} />*/}
+          {/*  <RemoveQuestionBankButton initialData={questionBank} />*/}
+          {/*</CardFooter>*/}
```
Comment (RealDyllon) [inline apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx]: can remove

File: apps/web/src/app/dashboard/teacher/quiz/_components/NewQuizButton.tsx [hunk 5]
```diff
@@ -44,43 +28,45 @@ import {
...
     no_of_questions: 0,
-    question_bank_id: 0,
     quiz_name: "",
     timer: 0
   });
-  const [maxQns, setMaxQns] = useState<number>(0);
-  const { data: questionsBanks } = trpc.questionBank.list.useQuery(undefined, {
-    initialData
-  });
+  const [maxQns] = useState<number>(initialData.questionsCount);
```
Comment (RealDyllon) [inline apps/web/src/app/dashboard/teacher/quiz/_components/NewQuizButton.tsx]: state is never set, so do we need a state hook? why not just replace maxQns with initialData.questionsCount

File: apps/web/src/app/dashboard/teacher/quiz/_components/NewQuizButton.tsx [hunk 6]
```diff
@@ -44,43 +28,45 @@ import {
...
+  };
 }
 interface quizProps {
   no_of_questions: number;
-  question_bank_id: number;
   quiz_name: string;
   timer: number;
 }
 export const NewQuizButton = ({ initialData }: Props) => {
+  console.log(initialData);
```
Comment (RealDyllon) [inline apps/web/src/app/dashboard/teacher/quiz/_components/NewQuizButton.tsx]: can remove

**Review by RealDyllon — CHANGES_REQUESTED** [review]

File: apps/web/src/app/dashboard/teacher/quiz/[slug]/_components/NewQuestionButton.tsx (hunk 1 above)
Comment (Zenteh-NTU) [reply apps/web/src/app/dashboard/teacher/quiz/[slug]/_components/NewQuestionButton.tsx]: Resolved in https://github.com/peiyuans-pals/toohakai/commit/19647d875eb27ab17a412917c24f814f0f436a1f

File: apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx (hunk 2 above)
Comment (Zenteh-NTU) [reply apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx]: Resolved in 
https://github.com/peiyuans-pals/toohakai/commit/19647d875eb27ab17a412917c24f814f0f436a1f

File: apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx (hunk 3 above)
Comment (Zenteh-NTU) [reply apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx]: Resolved in https://github.com/peiyuans-pals/toohakai/commit/19647d875eb27ab17a412917c24f814f0f436a1f

File: apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx (hunk 4 above)
Comment (Zenteh-NTU) [reply apps/web/src/app/dashboard/teacher/quiz/_components/Cards.tsx]: Resolved in https://github.com/peiyuans-pals/toohakai/commit/19647d875eb27ab17a412917c24f814f0f436a1f

File: apps/web/src/app/dashboard/teacher/quiz/_components/NewQuizButton.tsx (hunk 5 above)
Comment (Zenteh-NTU) [reply apps/web/src/app/dashboard/teacher/quiz/_components/NewQuizButton.tsx]: Resolved in https://github.com/peiyuans-pals/toohakai/commit/19647d875eb27ab17a412917c24f814f0f436a1f

File: apps/web/src/app/dashboard/teacher/quiz/_components/NewQuizButton.tsx (hunk 6 above)
Comment (Zenteh-NTU) [reply apps/web/src/app/dashboard/teacher/quiz/_components/NewQuizButton.tsx]: Resolved in https://github.com/peiyuans-pals/toohakai/commit/19647d875eb27ab17a412917c24f814f0f436a1f

**Pull request closed without merging**